.github/workflows/cloudflare-docs-preview.yml

```
@@ -0,0 +1,63 @@
+name: Deploy Docs Preview to Cloudflare Pages
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "docs/**"
+      - "mkdocs.yml"
+
+jobs:
+  deploy-preview:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      deployments: write
+      pull-requests: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install uv Package
+        run: |
+          pip install --upgrade pip
+          pip install uv==0.5.30
+
+      - name: Install Pre-commit in uv environment
+        run: uv run pip install pre-commit
+
+      - name: Install Dependencies
+        run: uv sync
```

pre-commit will be installed when uv sync is run
```suggestion
      - name: Install Dependencies
        run: uv sync
```